File: num_of_1Bit/num_of_1Bit.cpp
#include <iostream>
using namespace std;
class Solution
{
public:
	int func(unsigned n)
	{
		int count = 0;
		for (; n; n &= n-1)
		{
			count++;
			cout<<'c' <<count <<endl;
			cout<<'n'<< n <<endl;
		}
		return count;
	}
};

int main()
{
	Solution s;
	
	cout << s.func(6) <<endl;
	return 0;
}

File: num_of_1Bit/num_of_1Bit.py
class Solution:
	def num_of_Bits1(self, n):
		if n<0:
			print('please input the right num')
		else:
			string = bin(n)
			num = string.count('1')
		return num

if __name__ == '__main__':
	s = Solution()
	result = s.num_of_Bits1(6)
	print(result)
		
